pkg/internal/nodeinfo/cpu_models.go

```diff
@@ -0,0 +1,333 @@
+
+		for _, cpuModel := range cpuModels {
+			cpuModelCounts[cpuModel]++
+
+			if cpuCapacity != nil {
+				if cpuModelCores[cpuModel] == nil {
+					cpuModelCores[cpuModel] = resource.NewQuantity(0, resource.DecimalSI)
+				}
+				cpuModelCores[cpuModel].Add(*cpuCapacity)
+			}
+
+			if memoryCapacity != nil {
+				if cpuModelMemory[cpuModel] == nil {
+					cpuModelMemory[cpuModel] = resource.NewQuantity(0, resource.BinarySI)
+				}
+				cpuModelMemory[cpuModel].Add(*memoryCapacity)
+			}
+		}
+	}
+
+	newHash := generateHash(cpuModelCounts, cpuModelCores, cpuModelMemory)
+
+	cpuModelInfo.lock.Lock()
+	defer cpuModelInfo.lock.Unlock()
+
+	if cpuModelInfo.lastUpdateHash == newHash {
+		return false
+	}
+
+	cpuModelInfo.totalNodes = totalNodes
+	cpuModelInfo.totalCpu = totalCpu
+	cpuModelInfo.totalMemory = totalMemory
+	cpuModelInfo.models = make(map[string]*v1beta1.CpuModelInfo)
+	for cpuModel, count := range cpuModelCounts {
+		cpuStr := "0"
+		if cpuModelCores[cpuModel] != nil {
+			cpuStr = cpuModelCores[cpuModel].String()
+		}
+
+		memStr := "0"
+		if cpuModelMemory[cpuModel] != nil {
+			memStr = cpuModelMemory[cpuModel].String()
+		}
+
+		cpuModelInfo.models[cpuModel] = &v1beta1.CpuModelInfo{
+			Name:      cpuModel,
+			Benchmark: getCpuModelPassMarkScore(cpuModel),
+			Nodes:     count,
+			Cpu:       cpuStr,
+			Memory:    memStr,
+		}
+	}
+
+	cpuModelInfo.lastUpdateHash = newHash
+	return true
+}
+
+func extractCpuModels(labels map[string]string) []string {
+	cpuModels := sets.New[string]()
+
+	for key, value := range labels {
+		if strings.HasPrefix(key, CpuModelLabelPrefix) {
+			// Extract CPU model name from the label key
+			cpuModel := strings.TrimPrefix(key, CpuModelLabelPrefix)
+			// Only include if the value is "true" (meaning the node supports this model)
+			if value == "true" {
+				cpuModels.Insert(cpuModel)
+			}
+		}
+	}
+
+	return cpuModels.UnsortedList()
+}
+
+func getCpuModelPassMarkScore(cpuModel string) int {
+	// Try exact match first
+	if score, exists := cpuModelPassMarkScores[cpuModel]; exists {
+		return score
+	}
+
+	// Try to find a partial match for models with suffixes
+	for modelName, score := range cpuModelPassMarkScores {
+		if strings.Contains(cpuModel, modelName) || strings.HasPrefix(cpuModel, modelName) {
+			return score
+		}
+	}
+
+	// Return 0 for unknown models
+	return 0
+}
+
+func generateHash(counts map[string]int, cores map[string]*resource.Quantity, memory map[string]*resource.Quantity) string {
+	// Simple hash generation based on the data
+	var parts []string
+
+	models := make([]string, 0, len(counts))
+	for model := range counts {
+		models = append(models, model)
+	}
+
+	for _, model := range models {
+		cpuStr := "0"
+		if cores[model] != nil {
+			cpuStr = cores[model].String()
+		}
+
+		memStr := "0"
+		if memory[model] != nil {
+			memStr = memory[model].String()
+		}
+
+		part := model + ":" + strconv.Itoa(counts[model]) + ":" +
+			cpuStr + ":" + memStr
```

Comment:
IIRC, string concatenation is not so efficient. `fmt.Sprinf` is not great either, but more readable. 

so we can do:

```suggestion
		part := fmt.Sprintf("%s:%d:%s:%s", model, strconv.Itoa(counts[model]), cpuStr, memStr)
```

`strings.Builder` considered to be more efficient, but less readable; e.g.

```suggestion
	b := strings.Builder{}
	b.WriteString(model)
	b.WriteByte(':')
	...
	parts = append(parts, b.String())
```